packages/penrose-core/src/utils/Error.ts

```diff
@@ -249,17 +329,30 @@ export const parseError = (message: string): ParseError => ({
   message,
 });
 
-export const genericStyleError = (messages: string[]): PenroseError => ({
+// If there are multiple errors, just return the tag of the first one
+export const firstStyleError = (messages: StyleError[]): PenroseError => {
+  if (!messages.length) {
+    throw Error("internal error: expected at least one Style error");
+  }
+
+  return {
+    errorType: "StyleError",
+    tag: messages[0].tag, // COMBAK: is this confusing for showError?
+    messages: messages.map(showError),
+  }
+};
+
+export const genericStyleError = (messages: StyleError[]): PenroseError => ({
   errorType: "StyleError",
   tag: "GenericStyleError",
-  messages,
+  messages: messages.map(showError),
```

Comment:
`GenericStyleError` shouldn't exist anymore. Refer to my suggestion about `StyleErrorList`.